<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />



<meta name="date" content="2025-07-15" />

<title>Individual Analysis</title>

<script src="site_libs/header-attrs-2.30/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/flatly.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/pagedtable-1.1/css/pagedtable.css" rel="stylesheet" />
<script src="site_libs/pagedtable-1.1/js/pagedtable.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>



<style type="text/css">
  code {
    white-space: pre;
  }
  .sourceCode {
    overflow: visible;
  }
</style>
<style type="text/css" data-origin="pandoc">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  background-color: #f8f8f8; }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ef2929; } /* Alert */
code span.an { color: #8f5902; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #204a87; } /* Attribute */
code span.bn { color: #0000cf; } /* BaseN */
code span.cf { color: #204a87; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4e9a06; } /* Char */
code span.cn { color: #8f5902; } /* Constant */
code span.co { color: #8f5902; font-style: italic; } /* Comment */
code span.cv { color: #8f5902; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #8f5902; font-weight: bold; font-style: italic; } /* Documentation */
code span.dt { color: #204a87; } /* DataType */
code span.dv { color: #0000cf; } /* DecVal */
code span.er { color: #a40000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #0000cf; } /* Float */
code span.fu { color: #204a87; font-weight: bold; } /* Function */
code span.im { } /* Import */
code span.in { color: #8f5902; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #204a87; font-weight: bold; } /* Keyword */
code span.op { color: #ce5c00; font-weight: bold; } /* Operator */
code span.ot { color: #8f5902; } /* Other */
code span.pp { color: #8f5902; font-style: italic; } /* Preprocessor */
code span.sc { color: #ce5c00; font-weight: bold; } /* SpecialChar */
code span.ss { color: #4e9a06; } /* SpecialString */
code span.st { color: #4e9a06; } /* String */
code span.va { color: #000000; } /* Variable */
code span.vs { color: #4e9a06; } /* VerbatimString */
code span.wa { color: #8f5902; font-weight: bold; font-style: italic; } /* Warning */

</style>
<script>
// apply pandoc div.sourceCode style to pre.sourceCode instead
(function() {
  var sheets = document.styleSheets;
  for (var i = 0; i < sheets.length; i++) {
    if (sheets[i].ownerNode.dataset["origin"] !== "pandoc") continue;
    try { var rules = sheets[i].cssRules; } catch (e) { continue; }
    var j = 0;
    while (j < rules.length) {
      var rule = rules[j];
      // check if there is a div.sourceCode rule
      if (rule.type !== rule.STYLE_RULE || rule.selectorText !== "div.sourceCode") {
        j++;
        continue;
      }
      var style = rule.style.cssText;
      // check if color or background-color is set
      if (rule.style.color === '' && rule.style.backgroundColor === '') {
        j++;
        continue;
      }
      // replace div.sourceCode by a pre.sourceCode rule
      sheets[i].deleteRule(j);
      sheets[i].insertRule('pre.sourceCode{' + style + '}', j);
    }
  }
})();
</script>







<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Juvenile DA Analysis</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="diagram/interactive-flowchart.html">Analytics Architecture</a>
</li>
<li>
  <a href="data_processing.html">Data Processing</a>
</li>
<li>
  <a href="individual_analysis.html">Individual Analysis</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Population Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="population_analysis.html">Overview</a>
    </li>
    <li>
      <a href="weighted_LMMs.html">Weighted LMMs</a>
    </li>
    <li>
      <a href="balanced_resampling.html">Balanced Resampling</a>
    </li>
  </ul>
</li>
<li>
  <a href="reference.html">Reference</a>
</li>
<li>
  <a href="https://github.com/LXiao06/Juvenile_DA_analysis">GitHub</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Individual Analysis</h1>
<h4 class="date">2025-07-15</h4>

</div>


<p><br />
</p>
<div id="overview" class="section level2">
<h2>Overview</h2>
<p><strong>This tutorial demonstrates individual-level analysis of
processed fiber photometry dopamine data as well as song production from
juvenile songbirds. The analysis examines dopamine transient patterns,
peak amplitudes, and their relationship to vocal behavior development
using the VNS and ASAP packages.</strong><br />
<br />
</p>
<blockquote>
<h3 id="install-required-packages">Install Required Packages</h3>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">require</span>(remotes)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a>remotes<span class="sc">::</span><span class="fu">install_github</span>(<span class="st">&quot;LXiao06/ASAP&quot;</span>)</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>remotes<span class="sc">::</span><span class="fu">install_local</span>(<span class="fu">file.path</span>(<span class="fu">getwd</span>(),<span class="st">&quot;pkgs&quot;</span>, <span class="st">&quot;VNS_0.0.0.9001.tar.gz&quot;</span>), </span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a>                      <span class="at">dependencies =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
</blockquote>
</div>
<div id="load-packages" class="section level2">
<h2>Load Packages</h2>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;ASAP&quot;</span>)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a><span class="fu">library</span>(<span class="st">&quot;VNS&quot;</span>)</span></code></pre></div>
<p><br />
</p>
</div>
<div id="load-processed-data" class="section level2">
<h2>Load Processed Data</h2>
<p>Load the npm object created from the data processing pipeline.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a>npm <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&quot;./data/npm_B633.rds&quot;</span>)</span></code></pre></div>
<p><br />
</p>
</div>
<div id="visualize-song-development" class="section level2">
<h2>Visualize Song Development</h2>
<div id="figure-a" class="section level3">
<h3>Figure A</h3>
<p>Heatmap showing aligned amplitude envelopes of song recordings from
an example bird during development (72-91 dph). Each row represents one
epoch (50 epochs per dph). Dashed lines indicate motif onset and
offset.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" tabindex="-1"></a><span class="fu">plot_heatmap</span>(npm, <span class="at">segment_type =</span> <span class="st">&quot;bouts&quot;</span>,</span>
<span id="cb4-2"><a href="#cb4-2" tabindex="-1"></a>             <span class="at">balanced =</span>T,</span>
<span id="cb4-3"><a href="#cb4-3" tabindex="-1"></a>             <span class="at">window =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="fl">0.6</span>),</span>
<span id="cb4-4"><a href="#cb4-4" tabindex="-1"></a>             <span class="at">contrast =</span> <span class="dv">6</span>,</span>
<span id="cb4-5"><a href="#cb4-5" tabindex="-1"></a>             <span class="at">reference_lines =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">0.48</span>, <span class="fl">0.32</span>),</span>
<span id="cb4-6"><a href="#cb4-6" tabindex="-1"></a>             <span class="at">ylabel =</span> <span class="st">&quot;Day post hatch&quot;</span>,</span>
<span id="cb4-7"><a href="#cb4-7" tabindex="-1"></a>             <span class="at">verbose =</span> <span class="cn">FALSE</span>)</span></code></pre></div>
<p><img src="figs/FigA.png" alt="Figure A" /><br />
</p>
</div>
</div>
<div id="temporal-shift-of-dopamine" class="section level2">
<h2>Temporal Shift of Dopamine</h2>
<div id="figure-b" class="section level3">
<h3>Figure B</h3>
<p>Raster plot showing dopamine transients in individual epochs of the
example bird. Red and blue vertical lines indicate motif onset and
offset, respectively. Horizontal dashed lines delineate boundaries
between days.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" tabindex="-1"></a><span class="fu">plot_transient_raster</span>(npm,</span>
<span id="cb5-2"><a href="#cb5-2" tabindex="-1"></a>                      <span class="at">method =</span> <span class="st">&quot;epoch&quot;</span>, <span class="at">window =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>, <span class="fl">0.6</span>),</span>
<span id="cb5-3"><a href="#cb5-3" tabindex="-1"></a>                      <span class="at">region =</span> <span class="st">&quot;reg0&quot;</span>,</span>
<span id="cb5-4"><a href="#cb5-4" tabindex="-1"></a>                      <span class="at">ref_line =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">0.48</span>,<span class="fl">0.32</span>),</span>
<span id="cb5-5"><a href="#cb5-5" tabindex="-1"></a>                      <span class="at">point_size =</span> <span class="fl">0.3</span>,</span>
<span id="cb5-6"><a href="#cb5-6" tabindex="-1"></a>                      <span class="at">jitter_height =</span> <span class="fl">0.5</span>,</span>
<span id="cb5-7"><a href="#cb5-7" tabindex="-1"></a>                      <span class="at">ref_color =</span> <span class="fu">c</span>(<span class="st">&quot;red&quot;</span> , <span class="st">&quot;blue&quot;</span>),</span>
<span id="cb5-8"><a href="#cb5-8" tabindex="-1"></a>                      <span class="at">ylab =</span> <span class="st">&quot;Day post hatch&quot;</span>,</span>
<span id="cb5-9"><a href="#cb5-9" tabindex="-1"></a>                      <span class="at">relative_time_axis =</span> F)</span></code></pre></div>
<!-- ![Figure B](figs/FigB.png) -->
<p><img src="figs/FigB.png" /><!-- --></p>
<p><br />
</p>
</div>
<div id="figure-c" class="section level3">
<h3>Figure C</h3>
<p>Density ridge plots showing the distribution of centroid positions of
dopamine transients for each dph of the example bird. Red bars indicate
distribution peaks. Labels t₁-t₄ denote 200-ms segments spanning the
motif window.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a>segment_counts <span class="ot">&lt;-</span> <span class="fu">create_segment_table</span>(npm,</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a>                                       <span class="at">region =</span><span class="st">&quot;reg0&quot;</span>,</span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>                                       <span class="at">window =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">0.4</span>, <span class="fl">0.4</span>),</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a>                                       <span class="at">n_segments=</span> <span class="dv">12</span>)</span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a>transient_shift <span class="ot">&lt;-</span>  <span class="fu">analyze_timing_shift</span>(segment_counts,</span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a>                                         <span class="at">method =</span> <span class="st">&quot;centroid&quot;</span>,</span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a>                                         <span class="at">segment_duration_ms =</span> <span class="dv">67</span>,</span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a>                                         <span class="at">bandwidth =</span> <span class="dv">40</span>,</span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a>                                         <span class="at">highlight_position =</span><span class="st">&quot;mode&quot;</span>,</span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a>                                         <span class="at">xlab =</span> <span class="st">&quot;Centroid Position of DA Transients(ms)&quot;</span>)</span></code></pre></div>
<!-- ![Figure C](figs/FigC.png) -->
<p><img src="figs/FigC.png" /><!-- --></p>
<p><br />
</p>
</div>
</div>
<div id="amplitude-shift-of-dopamine" class="section level2">
<h2>Amplitude Shift of Dopamine</h2>
<div id="figure-l-m" class="section level3">
<h3>Figure L-M</h3>
<p>Representative example showing classic TD learning amplitude shift
pattern between early and late motif segments</p>
<p>(left) Peak amplitudes for early (t₁, red) and late (t₄, blue) time
windows across development. Early amplitude increased significantly
while late amplitude decreased significantly.</p>
<p>(right) Early/late amplitude ratio increased significantly across
development.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a>segment_tab <span class="ot">&lt;-</span> <span class="fu">create_segment_table</span>(npm,</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a>                                    <span class="at">region =</span> <span class="st">&quot;reg0&quot;</span>,</span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>                                    <span class="at">method =</span> <span class="st">&quot;epoch&quot;</span>,</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>                                    <span class="at">data_type =</span> <span class="st">&quot;peak_aligned&quot;</span>,</span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a>                                    <span class="at">window =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">0.4</span>, <span class="fl">0.4</span>),</span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>                                    <span class="at">n_segments =</span> <span class="dv">12</span>)</span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a></span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a>amplitude_shift <span class="ot">&lt;-</span> <span class="fu">analyze_amplitude_shift</span>(segment_tab,</span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a>                                   <span class="at">output =</span> <span class="st">&quot;data.frame&quot;</span>,</span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a>                                   <span class="at">segment_duration_ms =</span> <span class="dv">67</span>,</span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a>                                   <span class="at">segments_per_label =</span> <span class="dv">3</span>,      <span class="co"># Number of segments at each edge</span></span>
<span id="cb7-12"><a href="#cb7-12" tabindex="-1"></a>                                   <span class="at">label_pair =</span> <span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">4</span>) )        <span class="co"># First and last 200ms</span></span></code></pre></div>
<!-- ![Figure L-M](figs/FigL-M.png) -->
<p><img src="figs/FigL-M.png" /><!-- --></p>
<p><br />
</p>
</div>
<div id="supplemental-figure-1a-d" class="section level3">
<h3>Supplemental figure 1A-D</h3>
<p>Representative example showing amplitude redistribution across
different pairs of motif segment within 6 days’ sliding window</p>
<ol style="list-style-type: lower-alpha">
<li><p>Peak amplitudes for early (t₁, red) and late (t₃, blue) segments
across development. Early amplitude increased significantly while late
amplitude decreased significantly.</p></li>
<li><p>Early/late amplitude ratio (t₁/t₃) increased significantly across
development.</p></li>
<li><p>Peak amplitudes for early (t₂, red) and late (t₄, blue) segments
showing developmental trajectory. Early amplitude showed no significant
change while late amplitude decreased significantly.</p></li>
<li><p>Early/late amplitude ratio (t₂/t₄) increased significantly across
development.</p></li>
</ol>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" tabindex="-1"></a>amplitude_redistribution <span class="ot">&lt;-</span> <span class="fu">analyze_amplitude_redistribution</span>(segment_tab,</span>
<span id="cb8-2"><a href="#cb8-2" tabindex="-1"></a>                                                         <span class="at">output =</span> <span class="st">&quot;list&quot;</span>,</span>
<span id="cb8-3"><a href="#cb8-3" tabindex="-1"></a>                                                         <span class="at">segment_duration_ms =</span> <span class="dv">67</span>,</span>
<span id="cb8-4"><a href="#cb8-4" tabindex="-1"></a>                                                         <span class="at">segments_per_label =</span> <span class="dv">3</span>,</span>
<span id="cb8-5"><a href="#cb8-5" tabindex="-1"></a>                                                         <span class="at">step_size =</span> <span class="dv">2</span>)</span></code></pre></div>
<!-- ![Figure sup1](figs/Fig_sup1.png) -->
<p><img src="figs/Fig_sup1.png" /><!-- --></p>
<p><br />
</p>
</div>
</div>
<div id="information-theory-analysis" class="section level2">
<h2>Information Theory Analysis</h2>
<div id="figure-q-r" class="section level3">
<h3>Figure Q-R</h3>
<p>Representative example showing predictive coding development between
early and late motif segments.</p>
<p>(left) Forward conditional entropy H(t₄|t₁) (red) significantly
decreased with development, indicating improved predictability of late
events given early events. Backward conditional entropy H(t₁|t₄) (blue)
showed no significant change.</p>
<p>(right) Prediction asymmetry (H(t₁|t₄) - H(t₄|t₁)) significantly
increased across development.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" tabindex="-1"></a>segment_counts <span class="ot">&lt;-</span> <span class="fu">create_segment_table</span>(npm,</span>
<span id="cb9-2"><a href="#cb9-2" tabindex="-1"></a>                                       <span class="at">region =</span><span class="st">&quot;reg0&quot;</span>,</span>
<span id="cb9-3"><a href="#cb9-3" tabindex="-1"></a>                                       <span class="at">window =</span> <span class="fu">c</span>(<span class="sc">-</span><span class="fl">0.4</span>, <span class="fl">0.4</span>),</span>
<span id="cb9-4"><a href="#cb9-4" tabindex="-1"></a>                                       <span class="at">n_segments=</span><span class="dv">4</span>)</span>
<span id="cb9-5"><a href="#cb9-5" tabindex="-1"></a>dev_res <span class="ot">&lt;-</span> <span class="fu">analyze_directional_cH</span>(segment_counts,</span>
<span id="cb9-6"><a href="#cb9-6" tabindex="-1"></a>                                  <span class="at">source_seg =</span> <span class="dv">1</span>,</span>
<span id="cb9-7"><a href="#cb9-7" tabindex="-1"></a>                                  <span class="at">target_seg =</span> <span class="dv">4</span>)</span></code></pre></div>
<!-- ![Figure Q-R](figs/FigQ-R.png) -->
<p><img src="figs/FigQ-R.png" /><!-- --></p>
<p><br />
</p>
</div>
<div id="supplemental-figure-2a-d" class="section level3">
<h3>Supplemental figure 2A-D</h3>
<p>Representative example showing shift of directional information flow
across different pairs of motif segment within 6 days’ sliding
window</p>
<ol style="list-style-type: lower-alpha">
<li><p>Conditional entropies for forward (H(t₃|t₁), red) and backward
(H(t₁|t₃), blue) prediction across development. Forward entropy
decreased significantly indicating improved predictability of t₃ from
t₁, while backward entropy showed no significant change.</p></li>
<li><p>Prediction asymmetry (H(t₁|t₃) - H(t₃|t₁)) increased
significantly across development, with dominance crossover at DPH
74.4.</p></li>
<li><p>Conditional entropies for forward (H(t₄|t₂), red) and backward
(H(t₂|t₄), blue) prediction showing developmental trajectory. Forward
entropy decreased significantly while backward entropy showed no
significant change.</p></li>
<li><p>Prediction asymmetry (H(t₂|t₄) - H(t₄|t₂)) increased
significantly across development, with dominance crossover at DPH
78.7.</p></li>
</ol>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" tabindex="-1"></a>dev_res  <span class="ot">&lt;-</span> <span class="fu">analyze_directional_info_flow</span>(segment_counts, <span class="at">step_size =</span> <span class="dv">2</span>)</span></code></pre></div>
<!-- ![Figure sup2](figs/Fig_sup2.png) -->
<p><img src="figs/Fig_sup2.png" /><!-- --></p>
<p><br />
</p>
</div>
</div>
<div id="summary" class="section level2">
<h2>Summary</h2>
<p>This individual analysis successfully:</p>
<ol style="list-style-type: decimal">
<li><strong>Visualized</strong> song and dopamine transient patterns
across development</li>
<li><strong>Quantified</strong> timing and amplitude of dopamine
transients</li>
<li><strong>Analyzed</strong> temporal and amplitude shift of dopamine
activity within song motif</li>
<li><strong>Modeled</strong> developmental trajectories of dopamine
activity within song motif</li>
<li><strong>Generated</strong> comprehensive individual-level
reports</li>
<li><strong>Prepared</strong> results for population-level
analysis<br />
<br />
</li>
</ol>
</div>
<div id="session-info" class="section level2">
<h2>Session Info</h2>
<pre><code>## R version 4.4.1 (2024-06-14)
## Platform: x86_64-pc-linux-gnu
## Running under: Ubuntu 24.04.3 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.26.so;  LAPACK version 3.12.0
## 
## locale:
##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       
##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   
##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          
## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   
## 
## time zone: UTC
## tzcode source: system (glibc)
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## loaded via a namespace (and not attached):
##  [1] digest_0.6.39   R6_2.6.1        fastmap_1.2.0   xfun_0.54      
##  [5] cachem_1.1.0    knitr_1.50      htmltools_0.5.9 rmarkdown_2.30 
##  [9] lifecycle_1.0.4 cli_3.6.5       sass_0.4.10     jquerylib_0.1.4
## [13] compiler_4.4.1  tools_4.4.1     evaluate_1.0.5  bslib_0.9.0    
## [17] yaml_2.3.12     rlang_1.1.6     jsonlite_2.0.0</code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
